postr/apis/discord.py

```diff
@@ -0,0 +1,107 @@
+            return True
+        except Exception as exp:
+            print(f'Failed to send message: {exp}')
+            return False
+
+    def post_video(self, url: str, text: str) -> bool:
+        ''' This method takes in the url for the video the user
+        want to post and returns the success of this action'''
+        return False
+
+    def post_photo(self, url: str, text: str) -> bool:
+        ''' This method takes in the url for the photo the user
+        want to post and returns the success of this action'''
+        return False
+
+    def get_user_likes(self) -> int:
+        ''' This method returns the number of likes a user has'''
+        return -1
+
+    def get_user_followers(self, text: str) -> List[str]:
+        ''' This method returns a list of all the people that
+        follow the user'''
+        return None  # type: ignore
+
+    def remove_post(self, post_id: str) -> bool:
+        ''' This method removes the post with the specified id
+        and returns the successs of this action'''
+        return False
+
+    async def set_default_channel(self, channel_id: str) -> bool:
+        print('Tried to change default channel')
+        try:
+            await self.client.send_message(
+                client.get_channel(channel_id),
+                f'Default channel changed to <#{channel_id}>',
+            )
+            print(f'Default channel changed to {channel_id}')
+            self.default_channel = channel_id
+            return True
+        except Exception as e:
+            print(f'Failed to set default channel to channel with channel id={channel_id}')
+            print(e)
+            return False
+
+
+discord_api = Discord()
+client = discord_api.client
+# bot = commands.Bot(command_prefix='!', description="Command Parser")
+
+
+@client.event
+async def on_ready() -> None:
+    print(f'Logged in as{client.user.name}#{client.user.discriminator}')
+    print(f'ID: {client.user.id}')
+    # await discord_api.set_default_channel('499395129992413196')
+    await discord_api.post_text_async('Wow a test post!')
+
+
+@client.event
+async def on_message(message: discord.message) -> None:
+    print(f'New message received in {message.channel.name} from {message.author}: {message.content}')
+    if message.content.startswith('!channel'):
+        messages_in_channel = 0
+        tmp = await client.send_message(message.channel, 'Counting messages in channel...')
+        async for msg in client.logs_from(message.channel, limit=1000):
+            if msg:
+                messages_in_channel += 1
+        await client.edit_message(tmp, f'In total, {messages_in_channel} messages in this channel')
+    elif message.content.startswith('!rest'):
+        await client.send_message(message.channel, 'Resting for 5 seconds')
+        await asyncio.sleep(5)
+        await client.send_message(message.channel, 'Done resting')
+    else:
+        await client.add_reaction(message, '👍')
```

Comment:
Will this emoji break any windows users when looking at the code?

Reply:
🤔

Reply:
I think it'll just display as a box, which would be confusing but not a problem.

In next week's discord finalizing, I'll change all emojis to `\U+ABC123` format, just as a safeguard.